code/datums/components/religious_tool.dm

```diff
@@ -0,0 +1,149 @@
+	var/force_catalyst_afterattack = FALSE
+
+/datum/component/religious_tool/Initialize(_flags = ALL, _force_catalyst_afterattack = FALSE, override_catalyst_type)
+	. = ..()
+	SetGlobalToLocal() //attempt to connect on start in case one already exists!
+	operation_flags = _flags
+	force_catalyst_afterattack = _force_catalyst_afterattack
+	if(override_catalyst_type)
+		catalyst_type = override_catalyst_type
+
+/datum/component/religious_tool/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_RELIGION_NEWSECT, .proc/SetGlobalToLocal)
+	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY,.proc/AttemptActions)
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/on_examine)
+
+/datum/component/religious_tool/UnregisterFromParent()
+	UnregisterSignal(parent, list(COMSIG_PARENT_ATTACKBY,COMSIG_RELIGION_NEWSECT, COMSIG_PARENT_EXAMINE))
+
+/**
+  * Sets the easy access variable to the global if it exists.
+  */
+/datum/component/religious_tool/proc/SetGlobalToLocal()
+	if(!GLOB.religious_sect)
+		return
+	easy_access_sect = GLOB.religious_sect
+
+
+/**
+  * Since all of these involve attackby, we require mega proc. Handles Invocation, Sacrificing, And Selection of Sects.
+  */
+/datum/component/religious_tool/proc/AttemptActions(datum/source, obj/item/the_item, mob/living/user)
+	/**********Sect Selection**********/
+	if(!easy_access_sect)
+		if(!operation_flags & RELIGION_TOOL_SECTSELECT)
+			return
+		. = COMPONENT_NO_AFTERATTACK //At this point you're intentionally trying to select a sect.
+		if(user.mind.holy_role != HOLY_ROLE_HIGHPRIEST)
+			to_chat(user, "<span class='warning'>You are not the high priest, and therefore cannot select a religious sect.")
+			return
+		var/list/available_options = generate_available_sects(user)
+		if(!available_options)
+			return
+
+		var/sect_select = input(user,"Select a sect (You CANNOT revert this decision!)","Select a Sect",null) in available_options
+		if(!sect_select || !user.canUseTopic(parent, BE_CLOSE, FALSE, NO_TK))
+			to_chat(user,"<span class ='warning'>You cannot select a sect at this time.</span>")
+			return
+		var/type_selected = available_options[sect_select]
+		GLOB.religious_sect = new type_selected()
+		for(var/i in GLOB.player_list)
+			if(!isliving(i))
+				continue
+			var/mob/living/am_i_holy_living = i
+			if(!am_i_holy_living.mind?.holy_role)
+				continue
+			GLOB.religious_sect.on_conversion(am_i_holy_living)
+		easy_access_sect = GLOB.religious_sect
+		SEND_SIGNAL(parent, COMSIG_RELIGION_NEWSECT)
+		if(istype(parent,/obj/structure/altar_of_gods))
+			var/obj/structure/altar_of_gods/AOG = parent
+			if(easy_access_sect.altar_icon)
+				AOG.icon = easy_access_sect.altar_icon
+			if(easy_access_sect.altar_icon_state)
+				AOG.icon_state = easy_access_sect.altar_icon_state
+
+	/**********Rite Invocation**********/
+	else if(istype(the_item, catalyst_type))
```

Comment:
you cant just turn the bible into a var and expect to have fixed the problem!!! :^( if you want this to be only available to the bible make it a callback

Reply:
haha istype go brrrr

Reply:
Floyd is right, though, in that this should be handled as a callback.

Reply:
arent i always 😳

Reply:
(the answer is no)

Reply:
Not sure what you mean by this. this is the device HITTING parent, not parent itself (all religious tools have to be hit with a bible, but not all religious tools have to be a bible). Even if I converted to a callback, I would still have to specify this device is a bible to invoke the specific callback associated with the bible. I could maybe send a signal to the object hitting the device and hope for a return value after registering the signal with the bible although I don't see what we've gained by doing that.

I don't really see how this is different than say an inverse of the material_container component not accepting interactions by the object it's being attacked by if they aren't in a typecache (why not make a callback to items and see if they meet the pre-reqs?), or the archaeology component specifying the tool-type needs to be pickaxe/shovel (why not make a callback to all items, and have it default to seeing if it is a pickaxe/shoveling tool?).

**I'm happy to fix this if I can be directed to something that works similarly, as I've done with  converting the AoG Icon Changes to a callback.** The closest i've found (material container/archaeology) seems to suggest just ensuring the attacking object meets a specific set of parameters, including type, is sufficient for the current scope of my PR. Again, I'd like to consider the bible's functionality  a separate (although perhaps worthy to also be component'd) OOS for this PR, which is specific to the AoG functions.

Reply:
No, I see you have a point. Sorry for taking so long to answer.